app/controllers/Api/v1/posts_controller.rb

```diff
@@ -1,28 +1,23 @@
 class Api::V1::PostsController < ApplicationController
-  before_action :generate_token, only: :create
-  before_action :generate_salt, only: [:create]
-  before_action :destroy_exceed_posts, only: [:secret]
+  before_action :find_post, only: [:reveal, :secret]
 
   def create
     render json: {error: {error_type: "validation", error_message: "params missing"}}, status: 422 and return if post_params[:body].empty?
-    @post = Post.new(post_params)
-    @post.body = Ciphering.new(post_params[:body], post_params[:salty_password] || "").encrypt
-    @post.salty_password = @salt
-    @post.url_token =  @token
-    if @post.save
-      render json: {data: {url: @post.attributes}}, status: 200
+    @post = Post.create_new(post_params)
```

Comment:
why not call it create ? Since we're not inheriting from `Application::Record::Base` we will not clash with the ActiveRecord methods like `create`

Reply:
Ahh! will change it to just create